pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -491,6 +491,14 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 
 	pod := c.pods.getPodByIP(proxyIP)
 	if pod != nil {
+		// for split horizon EDS k8s multi cluster, in case there are pods of the same ip across clusters,
+		// which can happen when multi clusters using same pod cidr.
+		// As we have proxy Network meta, compare it with the network which endpoint belongs to,
+		// if they are not same, ignore the pod, because the pod is in another cluster.
+		if proxy.Metadata[model.NodeMetadataNetwork] != c.endpointNetwork(proxyIP) {
```

Comment:
does this default to "" when the network is not set on proxies (in single cluster mode)?

Reply:
I think this is a valid concern. What happens if my proxy is missing the network metadata (or alternatively endpointNetwork returns "" -- looking at the code this seems possible). In either one of these cases we could run into this code on single cluster.

Maybe we need to do this check only if both are `!= ""`? not sure if there is a better solution

Reply:
By default it is "" for both proxy and endpoint network.

Reply:
I see, thanks, so we shouldn't have this issue in single cluster